schematic/store/synapse.py

```diff
@@ -2188,70 +2267,73 @@ def _update_table_uuid_column(synStore: SynapseStorage, table_id: str, sg: Schem
                 # If there is not, then use the old `Uuid` column as a basis for the new `Id` column
                 else:
-                    # Create a new `Id` column based off of the old `Uuid` column, and store (column is empty)
-                    new_col = deepcopy(col)
-                    new_col['name'] = 'Id'
-                    schema.addColumn(new_col)
-                    schema = synStore.syn.store(schema)
-                
-                
-                    # Recently stored column is empty, so populated with uuid values
-                    TableOperations._populate_new_id_column(synStore, table_id, schema)
-
-                    # get the up-to-date table, remove old `Uuid` column, and store
-                    sleep(1)
-                    schema = synStore.syn.get(table_id)
-                    schema.removeColumn(col)
-                    schema = synStore.syn.store(schema)
+                    
+                    # Build ColumnModel that will be used for new column
+                    columnModelDict = {
+                        "id": None,
+                        "name": "Id",
+                        "defaultValue": None,
+                        "columnType": "STRING",
+                        "maximumSize": 64,
+                        "maximumListLength": 1,
+                    }
+
+                    # Send POST /column request to define new column and get new column ID
+                    newColResponse = self.synStore.send_api_request(
+                        request_type = "restPOST",
+                        uri = "https://repo-prod.prod.sagebase.org/repo/v1/column",
+                        body = columnModelDict,
+                    )
 
+                    # Define columnChange body
+                    columnChangeDict = {
+                        "concreteType": "org.sagebionetworks.repo.model.table.TableSchemaChangeRequest",
+                        "entityId": self.existingTableId,
+                        "changes": [
+                            {                        
+                                "oldColumnId": col['id'],
+                                "newColumnId": newColResponse['id'],
+                            }
+                        ]
+                    }
+
+                    # Build body for POST request
+                    schemaChangeBody = {
+                        "entityId": self.existingTableId,
+                        "changes": [columnChangeDict],
+                    }
+
+                    # Send POST request to change column name
+                    schemaChangeResponse = self.synStore.send_api_request(
+                        request_type = "restPOST",
+                        uri = f"https://repo-prod.prod.sagebase.org/repo/v1/entity/{self.existingTableId}/table/transaction/async/start",
+                        body = schemaChangeBody,
+                    )
                     # Exit iteration; only concerned with `Uuid` column
```

Comment:
I didn't realize this, but I dug deeper and I actually already have a function that does this: `syn._async_table_update` .  You can specify `wait=False` to speed up your code if you don't want to wait for the update to happen.


```suggestion
                   syn._async_table_update("syn51659783", changes=[columnChangeDict])
```